src/uwtools/config.py

```diff
@@ -342,178 +335,99 @@ def update_values(self, new_dict, dict_to_update=None):
-                        """ )
-                else:
-                    msg = dedent(f"""\
-
-                        ERROR:
-                        The input config file contains a constructor
-                        that is not registered with the uwtools package.
-
-                        constructor: {constructor}
-                        config file: {config_path}
-
-                        Define the constructor before proceeding.
-                        """)
-                self.log.exception(msg)
-                raise exceptions.UWConfigError(msg)
-
-        self.from_ordereddict(cfg)
-        return cfg
-
-    def dump_file(self, output_path):
-        ''' Write the dictionary to a YAML file '''
-
-        with open(output_path, 'w', encoding="utf-8") as file_name:
-            yaml.dump(self.data, file_name, sort_keys=False)
-
-    def _yaml_include(self, loader, node):
-        ''' Returns a dictionary that includes the contents of the referenced
-        YAML files, and is used as a contructor method for PyYAML'''
-
-        filepaths = loader.construct_sequence(node)
-        return self._load_paths(filepaths)
 
-    @property
-    def _yaml_loader(self):
-        ''' Set up the loader with the appropriate constructors. '''
-        loader = yaml.SafeLoader
-        loader.add_constructor('!INCLUDE', self._yaml_include)
-        return loader
-
-class F90Config(Config): #pylint: disable=unused-variable
+class F90Config(Config):
 
-    ''' Concrete class to handle Fortran namelist files. '''
+    """Concrete class to handle Fortran namelist files."""
 
-    def __init__(self, config_path=None, log_name=None):
-
-        ''' Load the file and update the dictionary '''
+    def __init__(self, config_path: Optional[str] = None, log_name: Optional[str] = None) -> None:
+        """Load the file and update the dictionary"""
         super().__init__(config_path, log_name)
 
         if config_path is not None:
             self.update(self._load())
             self.parse_include()
 
-    def _load(self, config_path=None):
-        ''' Load the user-provided Fortran namelist path into a dict
-        object. '''
+    def _load(self, config_path: Optional[str] = None) -> dict:
+        """Load the user-provided Fortran namelist path into a dict
+        object."""
+        # #PM# THIS SEEMS TO SILENTLY FAIL IF A NAMELIST IN THE FILE IS INVALID.
         config_path = config_path or self.config_path
-        with open(config_path, 'r', encoding="utf-8") as file_name:
-            cfg = f90nml.read(file_name).todict(complex_tuple=False)
-
-        cfg = dict(cfg)
-        self.from_ordereddict(cfg)
-        return cfg
+        # #PM# SINCE config_path in ctor IS ALSO OPTIONAL, THIS ASSERTION IS REQUIRED (PROBABLY BAD)
+        assert config_path is not None
+        with open(config_path, "r", encoding="utf-8") as f:
+            cfg = f90nml.read(f).todict(complex_tuple=False)
+        # #PM# WHY NOT JUST KEEP OrderedDict OBJECT?
```

Comment:
This was a requirement in the older versions we were testing. I believe we were seeing failures with Python 3.7 in our matrix of versions in GitHub Actions.

Reply:
@christinaholtNOAA {{OrderedDict}} dates from Python 3.2, so I'd be curious (but may never know) what was failing under 3.7.

In any case, and to ask a fundamental question: Python 3.8 is the earliest version that is not end-of-life. What's our stance on supporting EOL Pythons? IMO we should _not_ because they are not even getting security updates, and we should not encourage continued use of them. Should we have a policy of only tracking/testing currently-supported versions? If so, and assuming we can test successfully against 3.8-3.11, should we consider removing this {{OrderedDict}} -> {{dict}} translation? I could create a Jira ticket to explore/implement that...